Lib/cgi.py

```diff
@@ -483,7 +483,10 @@ def __enter__(self):
         return self
 
     def __exit__(self, *args):
-        self.file.close()
+        try:
+            self.file.close()
+        except AttributeError:
```

Comment:
I know this comes from `__del__`, but can we instead check whether `self.file` is `None` before calling `close` on it?

Reply:
I thought it was better to follow EAFP coding style rather than LBYL, do you have any consent why it should check whether `self.file` is `None`?